app/helpers/application_helper.rb

```diff
@@ -30,4 +30,23 @@ def file_size_for(file, decimal_places=2)
   def format_url_safe(text)
     ERB::Util.url_encode(text)
   end
-end
+
+  def any_fields_present(type, total)
+    if type == 'file'
```

Comment:
I took the liberty to make a couple of changes to the above, @spencerldixon, to make it more succinct. Hope that's okay 😀 
```ruby
def any_fields_present(type, total=5)
  file_fields_present = -> (n) { cms_block_content("file_#{n+1}".to_sym)&.file.present? }
  url_fields_present  = -> (n) { cms_block_content("url_#{n+1}".to_sym).present? }
  any_data_for_field = type == 'file' ? file_fields_present : url_fields_present
  
  (1..total).any? { |n| any_data_for_field[n] }
  # or even, although possibly less readable 
  # (1..total).any?(any_data_for_field)
end
```